Classes/Controller/SwitchUserController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the TYPO3 CMS project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+namespace KoninklijkeCollective\MyUserManagement\Controller;
+
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use TYPO3\CMS\Backend\Authentication\Event\SwitchUserEvent;
+use TYPO3\CMS\Backend\Utility\BackendUtility;
+use TYPO3\CMS\Core\SysLog\Type;
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+
+/**
+ * Provide modified version of "switchUserAction" to allow non-admin users to switch users
+ */
+class SwitchUserController extends \TYPO3\CMS\Backend\Controller\SwitchUserController
+{
+    public function switchUserAction(ServerRequestInterface $request): ResponseInterface
+    {
+        $currentUser = $this->getBackendUserAuthentication();
+        $targetUserId = (int)($request->getParsedBody()['targetUser'] ?? 0);
+
+        // Removed condition part "|| !$currentUser->isAdmin()" to allow non-admin users to switch user
+        if (!$targetUserId
+            || $targetUserId === (int)($currentUser->user[$currentUser->userid_column] ?? 0)
+            || $currentUser->getOriginalUserIdWhenInSwitchUserMode() !== null
+        ) {
+            return $this->jsonResponse(['success' => false]);
+        }
+
+        $targetUser = BackendUtility::getRecord('be_users', $targetUserId, '*', BackendUtility::BEenableFields('be_users'));
+        if ($targetUser === null || $targetUser['admin'] === 1 || $targetUser['disable'] === 1) {
+            return $this->jsonResponse(['success' => false]);
+        }
+
+        if (ExtensionManagementUtility::isLoaded('beuser')) {
+            // Set backend user listing module as starting module if installed
+            $currentUser->uc['startModuleOnFirstLogin'] = 'system_BeuserTxBeuser';
+        }

```

Comment:
What is the intention behind this? In my case this leads to troubles. When the non-admin user hits exist the user sees an "access denied". I am currently thinking what would be the best solution. Following come to mind:

1. Make `$currentUser->uc['startModuleOnFirstLogin'] = 'system_BeuserTxBeuser';` configurable.
2. Completely remove these lines.

However, for that it is helpful to understand the motivation for this condition.

Reply:
There is no special intention behind that. It's just a copy from what the core does: https://github.com/TYPO3/typo3/blob/main/typo3/sysext/backend/Classes/Controller/SwitchUserController.php#L80

So I think your solution no. 2 would be the best option. There is no need to keep these lines, imo.